File: ios/TelldusLiveWidget/WidgetCollectionViewCell.m
//
//  WidgetCollectionViewCell.m
//  TelldusLiveWidget
//
//  Created by Telldus Technologies AB on 28/05/18.
//  Copyright © 2018 Telldus Technologies AB. All rights reserved.
//

#import "WidgetCollectionViewCell.h"

@implementation WidgetCollectionViewCell

-(void)displayContent: (NSString *)label {
  _deviceLabel.text = label;
  _collectionViewData.layer.cornerRadius = 5.0;
}

-(void)updateCell: (int) index {
  [_onBtn setHidden:YES];
  [_offBtn setHidden:YES];
  [_downBtn setHidden:YES];
  [_upBtn setHidden:YES];
  [_stopBtn setHidden:YES];
  [_minusBtn setHidden:YES];
  [_plusBtn setHidden:YES];
  [_dimmerSlider setHidden:YES];
  [_dimmerSlider setUserInteractionEnabled: NO];
  
  _removeMinusBtn.layer.cornerRadius = 12.5;
  _removeMinusBtn.layer.zPosition = 1.0;
  _removeMinusBtn.tag = index;
}

@end

File: ios/TelldusDeviceLive/WidgetCollectionViewCell.swift
//
//  WidgetCollectionViewCell.swift
//  TelldusDeviceLive
//

import UIKit

class WidgetCollectionViewCell: UICollectionViewCell {
    
    @IBOutlet weak var deviceLabel: UILabel!
    @IBOutlet weak var btnView: UIView!
    @IBOutlet weak var onBtn: UIButton!
    @IBOutlet weak var offBtn: UIButton!
    @IBOutlet weak var downBtn: UIButton!
    @IBOutlet weak var upBtn: UIButton!
    @IBOutlet weak var stopBtn: UIButton!
    @IBOutlet weak var minusBtn: UIButton!
    @IBOutlet weak var plusBtn: UIButton!
    @IBOutlet weak var dimmerSlider: UISlider!
    @IBOutlet weak var collectionViewData: UIView!
    @IBOutlet weak var removeMinusBtn: UIButton!
    
    func displayContent(label: String) {
        deviceLabel.text = label
        collectionViewData.layer.cornerRadius = 5.0
    }
    
}
